OrbitCore/CrashHandler.cpp

```diff
@@ -64,6 +57,15 @@ CrashHandler::CrashHandler(const std::string& dump_path,
                                 /*asynchronous_start=*/false);
 }
 
+//-----------------------------------------------------------------------------
+void CrashHandler::SetUploadsEnabled(bool is_upload_enabled) const {
+  // set user preferences for dumps submission to collection server
+  if (crash_report_db_ != nullptr &&
+      crash_report_db_->GetSettings() != nullptr) {
+    crash_report_db_->GetSettings()->SetUploadsEnabled(is_upload_enabled);
```

Comment:
You're checking for nullptr and then you call GetSettings again which could in theory give you a different result.
Better store the settings pointer in a variable!